src/pipecat/processors/filters/stt_mute.py

```diff
@@ -0,0 +1,102 @@
+#
+# Copyright (c) 2024, Daily
+#
+# SPDX-License-Identifier: BSD 2-Clause License
+#
+
+from dataclasses import dataclass
+from enum import Enum
+from typing import Callable, Optional
+
+from loguru import logger
+
+from pipecat.frames.frames import (
+    BotStartedSpeakingFrame,
+    BotStoppedSpeakingFrame,
+    Frame,
+    StartInterruptionFrame,
+    StopInterruptionFrame,
+    STTMuteFrame,
+)
+from pipecat.processors.frame_processor import FrameDirection, FrameProcessor
+from pipecat.services.ai_services import STTService
+
+
+class STTMuteStrategy(Enum):
+    NEVER = "never"  # Never mute
+    FIRST_SPEECH = "first_speech"  # Mute only during first bot speech
+    ALWAYS = "always"  # Mute during all bot speech
+    CUSTOM = "custom"  # Allow custom logic via callback
+
+
+@dataclass
+class STTMuteConfig:
+    """Configuration for STTMuteProcessor"""
+
+    strategy: STTMuteStrategy = STTMuteStrategy.NEVER
+    # Optional callback for custom muting logic
+    should_mute_callback: Optional[Callable[["STTMuteProcessor"], bool]] = None
```

Comment:
Should the callback be awaitable? `Callable[["STTMuteProcessor"], Awaitable[bool]]`. The only drawback is that you can not pass a lambda, but nowadays I think this is fine. Better to be able to pass an `async` function with callbacks because you don't know what the user might want to do.